packages/x-date-pickers-pro/src/DateRangePickerDay/dateRangePickerDayClasses.ts

```diff
@@ -6,20 +6,32 @@ import {
 export interface DateRangePickerDayClasses {
   /** Styles applied to the root element. */
   root: string;
-  /** Styles applied to the root element if `isHighlighting=true` and `outsideCurrentMonth=false`. */
+  /** Styles applied to the root element if `isHighlighting=true`. */
   rangeIntervalDayHighlight: string;
   /** Styles applied to the root element if `isStartOfHighlighting=true`. */
   rangeIntervalDayHighlightStart: string;
   /** Styles applied to the root element if `isEndOfHighlighting=true`. */
   rangeIntervalDayHighlightEnd: string;
   /** Styles applied to the preview element. */
   rangeIntervalPreview: string;
-  /** Styles applied to the root element if `isPreviewing=true` and `outsideCurrentMonth=false`. */
+  /** Styles applied to the root element if `isPreviewing=true`. */
   rangeIntervalDayPreview: string;
-  /** Styles applied to the root element if `isStartOfPreviewing=true` or `day` is the start of the month. */
+  /** Styles applied to the root element if `isStartOfPreviewing=true`. */
   rangeIntervalDayPreviewStart: string;
-  /** Styles applied to the root element if `isEndOfPreviewing=true` or `day` is the end of the month. */
+  /** Styles applied to the root element if `isEndOfPreviewing=true`. */
   rangeIntervalDayPreviewEnd: string;
+  /** Styles applied to the root element if `outsideCurrentMonth=true` */
```

Comment:
This description also does not really match the fact that this class "Should" be applied to a selected range outside of the current month.
If we don't want or no longer need it, maybe change the class name to match it? 🤔